creds/tokens/credentials.go

```diff
@@ -11,16 +11,15 @@ import (
 	"github.com/nspcc-dev/neofs-s3-gw/api/cache"
 	"github.com/nspcc-dev/neofs-s3-gw/creds/accessbox"
 	cid "github.com/nspcc-dev/neofs-sdk-go/container/id"
-	"github.com/nspcc-dev/neofs-sdk-go/object/address"
 	oid "github.com/nspcc-dev/neofs-sdk-go/object/id"
 	"github.com/nspcc-dev/neofs-sdk-go/user"
 )
 
 type (
 	// Credentials is a bearer token get/put interface.
 	Credentials interface {
-		GetBox(context.Context, *address.Address) (*accessbox.Box, error)
-		Put(context.Context, *cid.ID, user.ID, *accessbox.AccessBox, uint64, ...*keys.PublicKey) (*address.Address, error)
+		GetBox(context.Context, oid.Address) (*accessbox.Box, error)
+		Put(context.Context, cid.ID, user.ID, *accessbox.AccessBox, uint64, ...*keys.PublicKey) (*oid.Address, error)
```

Comment:
Shouldn't we also use `oid.Address` as returned value instead of pointer?

Reply:
It depends on the approach taken: return `(T, error)` or `(*T, error)`. IMO first is expected to improve performance (but it's compilator-dependent) while latter is easier to read. We have same discussion in neofs-node https://github.com/nspcc-dev/neofs-node/pull/1418#discussion_r880087590.

I suggest you also discuss, if you decide to return by value - let me know and I'll change.

Reply:
I suggest to postpone decision about this and sync signatures in all places at once later.